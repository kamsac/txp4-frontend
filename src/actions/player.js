import dotenvConfiguration from '../dotenv-configuration';
import PlayerResourceMock from '../resources/player/mock';
import PlayerResource from '../resources/player';

export const RECEIVE_PLAYER = 'player:receivePlayer';
export function receivePlayer(player) {
  return {
    type: RECEIVE_PLAYER,
    payload: { player },
  };
}

function fetchPlayer() {
  const resource = dotenvConfiguration.API_URL ? PlayerResource : PlayerResourceMock;

  return (dispatch) => {
    resource.getPlayer()
      .then((response) => {
        dispatch(receivePlayer(response.data));
      });
  };
}

export const REQUEST_PLAYER = 'player:requestPlayer';
export function requestPlayer() {
  return (dispatch) => {
    dispatch(fetchPlayer());
  };
}
