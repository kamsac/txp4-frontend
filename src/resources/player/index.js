import axios from 'axios';
import dotenvConfiguration from '../../dotenv-configuration';

const PlayerResource = {
  getPlayer(playerLogin) {
    return axios.get(`${dotenvConfiguration.API_URL}/players/${playerLogin}`);
  },

  getInventory(playerLogin) {
    return axios.get(`${dotenvConfiguration.API_URL}/players/${playerLogin}/inventory`);
  },

  equipItem(playerLogin, itemId) {
    return axios.get(`${dotenvConfiguration.API_URL}/players/${playerLogin}/inventory/${itemId}/equip`);
  },

  unequipItem(playerLogin, itemId) {
    return axios.get(`${dotenvConfiguration.API_URL}/players/${playerLogin}/inventory/${itemId}/unequip`);
  },
};

export default PlayerResource;
