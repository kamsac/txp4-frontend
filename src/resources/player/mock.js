export const ITEMS = [
  {
    id: 'a1a1a1',
    name: 'Tire',
    vendorId: 'aaa',
    equipRegion: 'tire',
    tier: 3,
    isEquipped: true,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1.2,
      },
      {
        name: 'steering',
        value: 1.1,
      },
      {
        name: 'breaking',
        value: 1.2,
      },
      {
        name: 'grip',
        value: 1.5,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'b2b2b2',
    name: 'Less tired',
    vendorId: 'bbb',
    equipRegion: 'tire',
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 0.4,
      },
      {
        name: 'acceleration',
        value: 1,
      },
      {
        name: 'steering',
        value: 1.4,
      },
      {
        name: 'breaking',
        value: 2,
      },
      {
        name: 'grip',
        value: 3,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'c3c3c3',
    name: 'Shit engine',
    vendorId: 'ccc',
    equipRegion: 'engine',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 2,
      },
      {
        name: 'acceleration',
        value: 1.8,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1,
      },
      {
        name: 'grip',
        value: 0.2,
      },
      {
        name: 'gravity',
        value: 1.5,
      },
    ],
  },
  {
    id: 'd4d4d4',
    name: 'Some tire',
    vendorId: 'ddd',
    equipRegion: 'tire',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1.1,
      },
      {
        name: 'grip',
        value: 0.8,
      },
      {
        name: 'gravity',
        value: 0.9,
      },
    ],
  },
  {
    id: 'e5e5e5',
    name: 'Tire',
    vendorId: 'aaa',
    equipRegion: 'tire',
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1.2,
      },
      {
        name: 'steering',
        value: 1.1,
      },
      {
        name: 'breaking',
        value: 1.2,
      },
      {
        name: 'grip',
        value: 1.5,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'f6f6f6',
    name: 'Duper engine',
    vendorId: 'aaa',
    equipRegion: 'engine',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1.5,
      },
      {
        name: 'acceleration',
        value: 3,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1,
      },
      {
        name: 'grip',
        value: 1,
      },
      {
        name: 'gravity',
        value: 1.5,
      },
    ],
  },
  {
    id: 'f7f7f7',
    name: 'Tire',
    vendorId: 'aaa',
    equipRegion: 'tire',
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1.2,
      },
      {
        name: 'steering',
        value: 1.1,
      },
      {
        name: 'breaking',
        value: 1.2,
      },
      {
        name: 'grip',
        value: 1.5,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'g8g8g8',
    name: 'Less tired',
    vendorId: 'bbb',
    equipRegion: 'tire',
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 0.4,
      },
      {
        name: 'acceleration',
        value: 1,
      },
      {
        name: 'steering',
        value: 1.4,
      },
      {
        name: 'breaking',
        value: 2,
      },
      {
        name: 'grip',
        value: 3,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'h9h9h9',
    name: 'Shit engine',
    vendorId: 'ccc',
    equipRegion: 'engine',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 2,
      },
      {
        name: 'acceleration',
        value: 1.8,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1,
      },
      {
        name: 'grip',
        value: 0.2,
      },
      {
        name: 'gravity',
        value: 1.5,
      },
    ],
  },
  {
    id: 'i0i0i0',
    name: 'Some tire',
    vendorId: 'ddd',
    equipRegion: 'tire',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1.1,
      },
      {
        name: 'grip',
        value: 0.8,
      },
      {
        name: 'gravity',
        value: 0.9,
      },
    ],
  },
  {
    id: 'j1j1j1',
    name: 'Transmission',
    vendorId: 'ccc',
    equipRegion: 'transmission',
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 0.7,
      },
      {
        name: 'acceleration',
        value: 1.5,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1,
      },
      {
        name: 'grip',
        value: 1,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
];

export const PLAYER = {
  login: 'lymak',
  nick: '$s$o$a60l$c82y$fa3m$c82a$a60k',
  inventory: {
    items: ITEMS,
  },
};

const PlayerResourceMock = {
  getPlayer() {
    return new Promise((resolve) => {
      const response = {
        data: PLAYER,
      };
      setTimeout(() => resolve(response), 100);
    });
  },

  getInventory() {
    return new Promise((resolve) => {
      const response = {
        data: {
          items: ITEMS,
        },
      };
      setTimeout(() => resolve(response), 100);
    });
  },

  equipItem() {
    return new Promise((resolve) => {
      const response = {
        data: {
          items: ITEMS,
        },
      };
      setTimeout(() => resolve(response), 100);
    });
  },

  unequipItem() {
    return new Promise((resolve) => {
      const response = {
        data: ITEMS,
      };
      setTimeout(() => resolve(response), 100);
    });
  },
};

export default PlayerResourceMock;
