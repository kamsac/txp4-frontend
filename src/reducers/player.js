import inventoryReducer from './inventory';
import { RECEIVE_PLAYER } from '../actions/player';
import { EQUIP_ITEM, RECEIVE_INVENTORY, REQUEST_INVENTORY } from '../actions/inventory';

const initialState = {
  login: '',
  nick: '',
  inventory: {
    items: [],
  },
};

export default function playerReducer(state = initialState, action) {
  switch (action.type) {
    case RECEIVE_PLAYER:
      return Object.assign({}, state, action.payload.player);

    case REQUEST_INVENTORY:
    case RECEIVE_INVENTORY:
    case EQUIP_ITEM: {
      const inventory = inventoryReducer(state.inventory, action);
      return Object.assign({}, state, { inventory });
    }
    default:
      return state;
  }
}
