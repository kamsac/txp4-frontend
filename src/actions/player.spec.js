import {receivePlayer} from './player';

test('receivePlayer', () => {
  const somePlayer = {
    login: 'someLogin',
    nick: 'someNick',
    inventory: {
      items: []
    }
  };
  expect(receivePlayer(somePlayer)).toMatchSnapshot();
});
