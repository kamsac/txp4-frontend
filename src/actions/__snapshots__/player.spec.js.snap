// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`receivePlayer 1`] = `
Object {
  "payload": Object {
    "player": Object {
      "inventory": Object {
        "items": Array [],
      },
      "login": "someLogin",
      "nick": "someNick",
    },
  },
  "type": "player:receivePlayer",
}
`;
